src/test/regress/output/appendonly.source

```diff
@@ -1776,3 +1776,39 @@ LINE 1: explain select cmax, * from test_reporting_r;
 drop table test_reporting;
 drop table test_reporting_r;
 drop table test_reporting_c;
+-- Test AO functions with execution location on dispatcher.
+create table t1 (a int) with (appendoptimized=true, orientation=row);
+create table t2 (a int) with (appendoptimized=true, orientation=column);
+create table tables_list (tname text);
+insert into tables_list values ('t1'), ('t2');
+-- Queries below should fail with a proper error message.
+select get_ao_compression_ratio(tname) from tables_list;
+ERROR:  function 'get_ao_compression_ratio' must be executed on the dispatcher
+select get_ao_distribution(tname) from tables_list;
+ERROR:  function 'get_ao_distribution' must be executed on the dispatcher
+select gp_update_ao_master_stats(tname) from tables_list;
+ERROR:  function 'gp_update_ao_master_stats' must be executed on the dispatcher
+-- Query below is an example how to get the result
+-- intended by the failing query above.
+explain (costs off) select get_ao_compression_ratio(tname) from
+	unnest(array(select tname from tables_list)) as tname;
+                    QUERY PLAN                    
+--------------------------------------------------
+ Function Scan on unnest tname
+   InitPlan 1 (returns $0)  (slice2)
+     ->  Gather Motion 3:1  (slice1; segments: 3)
+           ->  Seq Scan on tables_list
+ Optimizer: Postgres query optimizer
+(5 rows)
+
+select get_ao_compression_ratio(tname) from
+	unnest(array(select tname from tables_list)) as tname;
+ get_ao_compression_ratio 
+--------------------------
+                        1
+                       -1
```

Comment:
Can you add explanation of this behavior to your note in the PR description? I feel like someone in the future could bump into this issue and try to look at this patch.

Reply:
Ok, added